dbms/src/Storages/transformQueryForExternalDatabase.cpp

```diff
@@ -81,9 +123,10 @@ String transformQueryForExternalDatabase(
       * copy only compatible parts of it.
       */
 
-    const ASTPtr & original_where = typeid_cast<const ASTSelectQuery &>(query).where_expression;
+    const ASTPtr & original_where = typeid_cast<const ASTSelectQuery &>(*clone_query).where_expression;
```

Comment:
Why we need `const` here If we change that variable inside `replaceConstFunction`?